oxidation/libparsec/crates/types/src/id.rs

```diff
@@ -188,24 +189,28 @@ impl std::fmt::Display for HumanHandle {
         write!(f, "{} <{}>", self.label, self.email)
     }
 }
+
 impl HumanHandle {
     pub fn new(email: &str, label: &str) -> Result<Self, &'static str> {
         let email = email.nfc().collect::<String>();
         let label = label.nfc().collect::<String>();
 
-        // TODO: how to check the email  easily ?
-        if email.is_empty() || email.len() >= 255 {
+        if !EmailAddress::is_valid(&email, None) || email.len() >= 255 {
             return Err("Invalid email address");
         }
-        if label.is_empty() || label.len() >= 255 {
+        // According to https://www.rfc-editor.org/rfc/rfc5322#section-3.2.3, these special characters are not allowed
```

Comment:
not allowed for what ?

the following comment indicate that the label is only informative, so why the restriction ?

```rust
// Label is purely informative
pub label: String,
```

Reply:
The label is equivalent to the display name when sending an email
And in test `[Test]` as label should lead to error

Reply:
So maybe the comment is misleading here (the one for the label field)